client/app/components/packages/applicant-fieldset.js

```diff
@@ -1,7 +1,32 @@
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
+import { STATE_OPTION_SET } from '../../models/applicant';
 
 export default class ApplicantFieldset extends Component {
+  get stateOptions() {
+    return Object.values(STATE_OPTION_SET);
+  }
+
+  // TODO: Use a global helper instead after writing
```

Comment:
Why not just do it now? It could be a helper called "group-by". Here's how D3 describes it: https://github.com/d3/d3-array#group.

```handlebars
  {{group-by 'code' this.stateOptions}}
```

Reply:
Perhaps we could have another helper that looks up optionsets for us:

```handlebars
{{group-by 'code' (lookup-options 'applicants' 'state')}}
```

Reply:
https://github.com/DockYard/ember-composable-helpers#group-by ?

Reply:
Thinking of a helper with this API, what do you think? 
@trbmcginnis @bfreeds 

**{{optionset 'applicant' 'state'}}** --> returns the optionset object
```js
    { OR: { label: 'OR', code: 12345}, { label: 'MN', code: 123456 } ... } 
```

**{{optionset 'applicant' 'state' 'list'}}** --> returns an optionset options as an array
```js
    [ { label: 'OR', code: 12345} ... }, { label: 'MN', code: 123456 } ... ] 
```

**{{optionset  'applicant' 'state' 'code' 'OR'}}** --> returns code for given label
```
12345
```

**{{optionset  'applicant' 'state' 'label' 12345}}** --> returns label for given code
```
'OR'
```

Reply:
That looks awesome!

Reply:
@godfreyyeung I love your proposed interface for working with changesets -- that would be really pleasant & ergonomic to work with.